Adopt more smart pointers in Navigator

rwlbuis opened the pull request:
#### bb1d80ceb53e75aa8c57c8515cf017c6c04b0e8f
<pre>
Adopt more smart pointers in Navigator
<a href="https://bugs.webkit.org/show_bug.cgi?id=273547">https://bugs.webkit.org/show_bug.cgi?id=273547</a>

Reviewed by Chris Dumez and Tim Nguyen.

Adopt more smart pointers in Navigator based on the
[alpha.webkit.UncountedLocalVarsChecker] warning.

* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::userAgent const):
(WebCore::Navigator::platform const):
(WebCore::Navigator::share):
(WebCore::Navigator::showShareData):
(WebCore::Navigator::initializePluginAndMimeTypeArrays):
(WebCore::Navigator::cookieEnabled const):
(WebCore::Navigator::setAppBadge):
(WebCore::Navigator::setClientBadge):

Canonical link: <a href="https://commits.webkit.org/278553@main">https://commits.webkit.org/278553@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b660eb1d920c470f446bd001e1a33b26093d8cab

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [⏳ 🧪 style ](https://ews-build.webkit.org/#/builders/Style-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios ](https://ews-build.webkit.org/#/builders/iOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wincairo ](https://ews-build.webkit.org/#/builders/Win-Build-EWS "Waiting in queue, processing has not started yet") 
| [⏳ 🧪 bindings ](https://ews-build.webkit.org/#/builders/Bindings-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 ios-sim ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 mac-AS-debug ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 wpe-wk2 ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitperl ](https://ews-build.webkit.org/#/builders/WebKitPerl-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2 ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-mac ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-wpe ](https://ews-build.webkit.org/#/builders/WPE-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 webkitpy ](https://ews-build.webkit.org/#/builders/WebKitPy-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 ios-wk2-wpt ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk1 ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc ](https://ews-build.webkit.org/#/builders/JSC-Tests-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-ios ](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🛠 🧪 jsc-arm64 ](https://ews-build.webkit.org/#/builders/JSC-Tests-arm64-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv ](https://ews-build.webkit.org/#/builders/tvOS-17-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-AS-debug-wk2 ](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Debug-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 gtk-wk2 ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [⏳ 🧪 services ](https://ews-build.webkit.org/#/builders/Services-EWS "Waiting in queue, processing has not started yet") | [⏳ 🛠 tv-sim ](https://ews-build.webkit.org/#/builders/tvOS-17-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 mac-wk2-stress ](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-Build-EWS "Waiting in queue, processing has not started yet") | [⏳ 🧪 api-gtk ](https://ews-build.webkit.org/#/builders/GTK-Build-EWS "Waiting in queue, processing has not started yet") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11156 "Built successfully and passed tests") | [⏳ 🛠 watch ](https://ews-build.webkit.org/#/builders/watchOS-10-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🛠 jsc-armv7 ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [⏳ 🛠 watch-sim ](https://ews-build.webkit.org/#/builders/watchOS-10-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | | [⏳ 🧪 jsc-armv7-tests ](https://ews-build.webkit.org/#/builders/JSC-ARMv7-32bits-Build-EWS "Waiting in queue, processing has not started yet") | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/page/Navigator.cpp:
```diff
@@ -81,7 +81,7 @@ Navigator::~Navigator() = default;
 String Navigator::appVersion() const
 {
-    auto* frame = this->frame();
+    SUPPRESS_UNCOUNTED_LOCAL auto* frame = this->frame();
```
I don't think we should use SUPPRESS_UNCOUNTED_LOCAL

rwlbuis replied:
Removed.

cdumez commented on Source/WebCore/page/Navigator.cpp:
```diff
@@ -333,21 +333,21 @@ bool Navigator::pdfViewerEnabled()
 bool Navigator::cookieEnabled() const
 {
-    auto* frame = this->frame();
+    SUPPRESS_UNCOUNTED_LOCAL auto* frame = this->frame();
```
ditto.

rwlbuis replied:
Ditto.

cdumez commented on Source/WebCore/page/Navigator.cpp:
```diff
@@ -389,31 +389,31 @@ GPU* Navigator::gpu()
 Document* Navigator::document()
 {
-    auto* frame = this->frame();
+    SUPPRESS_UNCOUNTED_LOCAL auto* frame = this->frame();
```
ditto.

rwlbuis replied:
Also removed.

rwlbuis committed:
Adopt more smart pointers in Navigator
https://bugs.webkit.org/show_bug.cgi?id=273547

Reviewed by Chris Dumez and Tim Nguyen.

Adopt more smart pointers in Navigator based on the
[alpha.webkit.UncountedLocalVarsChecker] warning.

* Source/WebCore/page/Navigator.cpp:
(WebCore::Navigator::userAgent const):
(WebCore::Navigator::platform const):
(WebCore::Navigator::share):
(WebCore::Navigator::showShareData):
(WebCore::Navigator::initializePluginAndMimeTypeArrays):
(WebCore::Navigator::cookieEnabled const):
(WebCore::Navigator::setAppBadge):
(WebCore::Navigator::setClientBadge):

Canonical link: https://commits.webkit.org/278553@main
